pkg/helpers/helpers.go

```diff
@@ -887,11 +916,126 @@ func FeatureGatesArgs(featureGates []operatorapiv1.FeatureGate, component string
 }
 
 func isValidFeatureGate(feature string, component string) bool {
-	if featureGates, ok := FeatureGatesMap[component]; ok {
+	if featureGates, ok := featureGatesMap[component]; ok {
 		if _, ok := featureGates[featuregate.Feature(feature)]; ok {
 			return true
 		}
 	}
 
 	return false
 }
+
+func defaultNilFeatureGates(componentKey string) []string {
+	return defaultNilFeatureGatesMap[componentKey]
+}
+
+func getComponentKey(operatorType string, componentName string) string {
+	var componentKey string
+	switch componentName {
+	case componentNameRegistration:
+		if operatorType == OperatorTypeClusterManager {
+			componentKey = componentKeyHubRegistration
+		}
+		if operatorType == OperatorTypeKlusterlet {
+			componentKey = componentKeySpokeRegistration
+		}
+	case componentNameWork:
+		if operatorType == OperatorTypeClusterManager {
+			componentKey = componentKeyHubWork
+		}
+		if operatorType == OperatorTypeKlusterlet {
+			componentKey = componentKeySpokeWork
+		}
+	}
+
+	return componentKey
+}
+
+func getConditionType(componentName string) string {
+	var conditionType string
+	switch componentName {
+	case componentNameRegistration:
+		conditionType = RegistrationFeatureGatesValid
```

Comment:
hrm, I am thinking why we need to conditions. Shouldn't we have one condition and different reasons?

Reply:
You mean merge the RegistrationFeatureGatesValid and WorkFeatureGatesValid into one condition？

Reply:
yes, I do not see why we need two conditions.